pkg/controller/kogitoinfra/mongodb.go

```diff
@@ -0,0 +1,308 @@
+	envProps = append(envProps, framework.CreateSecretEnvVar(propertiesMongoDBQuarkus[envVarMongoDBPassword], mongoDBSecretName, infrastructure.MongoDBAppSecretPasswordKey))
+	envProps = append(envProps, framework.CreateSecretEnvVar(propertiesMongoDBSpring[envVarMongoDBDatabase], mongoDBSecretName, infrastructure.MongoDBAppSecretDatabaseKey))
+	envProps = append(envProps, framework.CreateSecretEnvVar(propertiesMongoDBQuarkus[envVarMongoDBDatabase], mongoDBSecretName, infrastructure.MongoDBAppSecretDatabaseKey))
+	return envProps, nil
+}
+
+func getMongoDBAppProps(cli *client.Client, mongoDBInstance *mongodb.MongoDB) (map[string]string, error) {
+	appProps := map[string]string{}
+
+	mongoDBURI := mongoDBInstance.Status.MongoURI
+	if len(mongoDBURI) > 0 {
+		mongoDBParsedURL, err := url.ParseRequestURI(mongoDBURI)
+		if err != nil {
+			return nil, err
+		}
+		appProps[propertiesMongoDBSpring[appPropMongoDBHost]] = mongoDBParsedURL.Hostname()
+		appProps[propertiesMongoDBSpring[appPropMongoDBPort]] = mongoDBParsedURL.Port()
+		appProps[propertiesMongoDBQuarkus[appPropMongoDBURI]] = mongoDBURI
+	}
+
+	return appProps, nil
+}
+
+func updateMongoDBAppPropsInStatus(cli *client.Client, mongoDBInstance *mongodb.MongoDB, instance *v1alpha1.KogitoInfra) error {
+	log.Debugf("going to Update MongoDB app properties in kogito infra instance status")
+	appProps, err := getMongoDBAppProps(cli, mongoDBInstance)
+	if err != nil {
+		return err
+	}
+	instance.Status.AppProps = appProps
+	log.Debugf("Following app properties are set infra status : %s", appProps)
+	return nil
+}
+
+func updateMongoDBEnvVarsInStatus(cli *client.Client, mongoDBInstance *mongodb.MongoDB, instance *v1alpha1.KogitoInfra, scheme *runtime.Scheme) error {
+	log.Debugf("going to Update MongoDB env properties in kogito infra instance status")
+	envVars, err := getMongoDBSecretEnvVars(cli, mongoDBInstance, instance, scheme)
+	if err != nil {
+		return err
+	}
+	instance.Status.Env = envVars
+	log.Debugf("Following env properties are set infra status : %s", envVars)
+	return nil
+}
+
+func loadDeployedMongoDBInstance(cli *client.Client, instanceName string, namespace string) (*mongodb.MongoDB, error) {
+	log.Debug("fetching deployed kogito mongoDB instance")
+	mongoDBInstance := &mongodb.MongoDB{}
+	if exits, err := kubernetes.ResourceC(cli).FetchWithKey(types.NamespacedName{Name: instanceName, Namespace: namespace}, mongoDBInstance); err != nil {
+		log.Error("Error occurs while fetching kogito mongoDB instance")
+		return nil, err
+	} else if !exits {
+		log.Debug("Kogito mongoDB instance is not exists")
+		return nil, nil
+	} else {
+		log.Debug("Kogito mongoDB instance found")
+		return mongoDBInstance, nil
+	}
+}
+
+func createNewMongoDBInstance(cli *client.Client, name string, namespace string, instance *v1alpha1.KogitoInfra, scheme *runtime.Scheme) (*mongodb.MongoDB, error) {
```

Comment:
I think we need to stay away from this approach. The others are just kept there for backward compatibility, should go soon. Users must create their references, we just bound to them.

Reply:
In Knative use case, for instance, we do not create the `Broker`. It's a resource we expect from the user since it has its subtleties that are tied to a particular scenario.

Reply:
removing